### PR TITLE
CMake: Add support for macOS bundles

### PR DESCRIPTION
**Pull request by Holzhaus**
i-have-no-idea-what-im-doing-dog-feat-1-620x400

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1208,6 +1244,32 @@ set(CPACK_WIX_PROPERTY_ARPHELPLINK "${CPACK_PACKAGE_HOMEPAGE_URL}")
 set(CPACK_WIX_UI_BANNER "${CMAKE_CURRENT_SOURCE_DIR}/build/wix/images/banner.bmp")
 set(CPACK_WIX_UI_DIALOG "${CMAKE_CURRENT_SOURCE_DIR}/build/wix/images/dialog.bmp")
 
+set(CPACK_BUNDLE_NAME "Mixxx")
+set(CPACK_BUNDLE_ICON "${CMAKE_CURRENT_SOURCE_DIR}/res/osx/VolumeIcon.icns")
+set(CPACK_BUNDLE_APPLE_ENTITLEMENTS "${CMAKE_CURRENT_SOURCE_DIR}/build/osx/entitlements.plist")
+set(MACOS_BUNDLE_VERSION "${CMAKE_PROJECT_VERSION}")
+set(MACOS_BUNDLE_SHORTVERSION "${CMAKE_PROJECT_VERSION}")
+set(MACOS_BUNDLE_MINOSVERSION "10.11.0")
```
Comment (Be-ing) [inline CMakeLists.txt:1252]: I thought the minimum macOS version was 10.13?

Comment (Holzhaus) [reply CMakeLists.txt:1252]: Apparently not, I took that from https://github.com/mixxxdj/mixxx/blob/master/build/osx/product_definition.plist#L11, which is parsed here: https://github.com/mixxxdj/mixxx/blob/master/build/util.py#L194-L206

Comment (Be-ing) [reply CMakeLists.txt:1252]: Hmm, I'm not sure why I was thinking 10.13. https://mixxx.org/download/ says 10.11 as well. IIRC the Jenkins server runs 10.13?

**Comment by Holzhaus** [comment]
I give up. I can't implement for an OS that I never used and that I can't test locally. If someone wants to take over, feel free to do so.

**Comment by ferranpujolcamins** [comment]
@Holzhaus Thanks for having a look at this! I'll try to take over.

**Pull request closed without merging**